rita_client/src/operator_update/mod.rs

```diff
@@ -168,48 +169,69 @@ pub async fn operator_update(ops_last_seen_usage_hour: Option<u64>) -> u64 {
         }
     };
     let last_seen_hour = ops_last_seen_usage_hour.unwrap_or(0);
-    let mut hours_to_send: u64 = 0;
-    // first check whats the last saved hr, send everything from that hr on
-    // but, we need to save a var of the last x hrs not seen, and load that up with the next checkin cycle if >0.
-    if current_hour - last_seen_hour > 0 {
-        hours_to_send = current_hour - last_seen_hour;
-    }
+    // We check that the difference is >1 because we leave a 1 hour buffer to prevent from sending over an incomplete current hour
+    let send_hours = current_hour - last_seen_hour > 1;
     let mut usage_tracker_data: Option<UsageTracker> = None;
-    // only deal with this if we actually do need to send some hours
     // if ops_last_seen_usage_hour is a None the thread has restarted and we are waiting for ops to tell us how much
-    // data we need to send, which will be populated with the next checkin cycle
-    if hours_to_send != 0 && ops_last_seen_usage_hour.is_some() {
+    // data we need to send, which will be populated with the next checkin cycle. 730 is the average numbers of hours
+    // in a month, and we only send 1 month at a time if ops is requesting the full usage history.
+    if send_hours && ops_last_seen_usage_hour.is_some() {
```

Comment:
this should all be broken out into a function for easier unit testing + organization this is a big if block.

Reply:
way compressed the block but this can be broken out if it's still too big a chunk

Reply:
you shouldn't be running any of the loops in here. That should be in it's own function to be tested.